File: Programming EF Code First/chapter3/Chapter 3 VB/Data Annotations/Model/Address.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  <ComplexType>
  Public Class Address
	Public Property AddressId() As Integer
	<MaxLength(150)>
	Public Property StreetAddress() As String
	Public Property City() As String
	Public Property State() As String
	Public Property ZipCode() As String
  End Class
End Namespace

File: Programming EF Code First/chapter4/Chapter4/BAGA_DataAnnotations/Model/Address.cs
using System.ComponentModel.DataAnnotations;

namespace Model
{
  [ComplexType]
  public class Address
  {
    public int AddressId { get; set; }
    [MaxLength(150)]
    public string StreetAddress { get; set; }
    public string City { get; set; }
    public string State { get; set; }
    public string ZipCode { get; set; }
  }
}